### ISSUE TITLE
PeakGroup results cause a `NoAttribute` Exception when an animal has no tracer specified

### ISSUE THREAD

**Issue by lparsons** [Princeton-LSI-ResearchComputing/tracebase#316]
### Problem

Viewing the PeakGroups for studies where some animals have no tracer specified causes an `AttributeError` Excpetion.

The animal `MN_exp027e_M1` was loaded incorrectly without a tracer. It appears that the tracer should be `arginine`. It's possible (likely) the other animals in the study were also loaded incorrectly. However, we have determined that TraceBase *should* support Animals with no tracer (see #325).

### Steps to reproduce

1. Load the MN exp027e KR infusion study - see https://github.com/PrincetonUniversity/tracebase-rabinowitz-data/pull/4

*Note:* This should be more easily reproduced when #325 is completed.

This was discovered attempting to view PeakGroups for the (incorrectly loaded) "MN_exp027e_acute KR infusion" study on tracebase-dev, which caused a `NoReverseMatch` exception.

1. View study list - http://tracebase-dev.princeton.edu/DataRepo/studies/
2. View Study "MN_exp027e_acute KR infusion" - http://tracebase-dev.princeton.edu/DataRepo/studies/3/
3. Click on link to view PeakGroup data - http://tracebase-dev.princeton.edu/DataRepo/search_basic/Study/id/exact/3/peakgroups/

### Current behavior

```
Environment:


Request Method: GET
Request URL: http://tracebase-dev.princeton.edu/DataRepo/search_basic/Study/id/exact/3/peakgroups/

Django Version: 3.2.5
Python Version: 3.8.8
Installed Applications:
['DataRepo.apps.DatarepoConfig',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']


Template error:
In template /var/www/tracebase/DataRepo/templates/DataRepo/search/results/peakgroups.html, error at line 176
   'NoneType' object has no attribute 'id'
   166 :                             <!-- Enrichment Abundance -->
   167 :                             <td class="text-end">
   168 :                                 <p title="{{ pg.enrichment_abundance|floatformat:10 }}">{{ pg.enrichment_abundance|floatformat:4 }}</p>
   169 :                             </td>
   170 : 
   171 :                             <!-- Normalized Labeling -->
   172 :                             <td class="text-end">
   173 :                                 {% if pg.normalized_labeling is None %}
   174 :                                     <p title="Associated serum sample not found.">None</p>
   175 :                                 {% else %}
   176 :                                     <p title=" {{ pg.normalized_labeling|floatformat:15 }} ">{{ pg.normalized_labeling|floatformat:4 }}</p>
   177 :                                 {% endif %}
   178 :                             </td>
   179 : 
   180 :                             <!-- Peak Group Set Filename -->
   181 :                             <td>
   182 :                                 <!-- Put displayed link text first for sorting -->
   183 :                                 <div style="display:none;">{{ pg.peak_group_set.filename }}</div>
   184 :                                 <a href="{% url 'peakgroupset_detail' pg.peak_group_set.id %}">
   185 :                                     {{ pg.peak_group_set.filename }}
   186 :                                 </a>


Traceback (most recent call last):
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 829, in _resolve_lookup
    current = current[bit]

During handling of the above exception ('PeakGroup' object is not subscriptable), another exception occurred:
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/var/www/tracebase/DataRepo/views.py", line 266, in search_basic
    return render(
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/shortcuts.py", line 19, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 170, in render
    return self._render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/loader_tags.py", line 195, in render
    return template.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 172, in render
    return self._render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/loader_tags.py", line 53, in render
    result = self.nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/loader_tags.py", line 195, in render
    return template.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 172, in render
    return self._render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/loader_tags.py", line 53, in render
    result = self.nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/defaulttags.py", line 211, in render
    nodelist.append(node.render_annotated(context))
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/defaulttags.py", line 211, in render
    nodelist.append(node.render_annotated(context))
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 988, in render
    output = self.filter_expression.resolve(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 671, in resolve
    obj = self.var.resolve(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 796, in resolve
    value = self._resolve_lookup(context)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/template/base.py", line 837, in _resolve_lookup
    current = getattr(current, bit)
  File "/usr/local/tracebase/lib/python3.8/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/var/www/tracebase/DataRepo/models.py", line 1023, in normalized_labeling
    .filter(compounds__id=self.msrun.sample.animal.tracer_compound.id)

Exception Type: AttributeError at /DataRepo/search_basic/Study/id/exact/3/peakgroups/
Exception Value: 'NoneType' object has no attribute 'id'
```

### Expected behavior

A list of PeakGroup records belonging to that Study

### Suggested Change

Some problem attempting to calculate the normalized labeling (some value is `None` and thus has no `id` attribute).

### Comment

Any other relevant context or links to what you were working on when you
discovered the problem.

-----

### Assumptions

- List of assumptions made WRT the code
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- List of conditions to be met for the feature
- E.g. Every column/row must display a value, i.e. cannot be empty

### Limitations

- A list of things this work will not fix
- E.g. Getting edge case X to work requires too much effort and will not be
  fixed in this effort.

### Affected/Changed Components

- Files
- Environment variables
- External executables
- Database tables
- Cron job settings
- Etc.

#### GUI Change description

Describe changes the user will see.

#### Code Change Description *(Pseudocode optional)*

Describe code changes planned for the fix.

#### Tests

A test should be planned for each requirement (above), where possible.

**Comment by lparsons** [comment]
I believe this is likely due to #312

**Comment by hepcat72** [comment]
Confirmed.  This is due to #312.  `tracer_compound` resolves to None and `.id` thus errors out.  I presume that this is a `wontfix` in deference to the loading issue.

**Comment by lparsons** [comment]
We have determined that this scenario should indeed be supported, and thus this is an issue that should be fixed, ideally it would be tested after #325 is completed.

**Comment by hepcat72** [comment]
Looks like this is more complicated than I initially thought.  Found a problem in the `normalized_labeling` cached property that indexes a tracer `id` that doesn't exist...
